### PR TITLE
[2.0.0-1] Deploy all the APIs from Operator to Adapter using a Rest server

### PR DESCRIPTION
**Pull request by Jayanie**
## Purpose

- This PR ports the changes included in #507

### CODE REVIEW

File: api-operator/pkg/envoy/server/api/restserver/configure_restapi.go [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+				logRestServer.Error(err, "Error when writing zipped file to response")
+				return err
+			}
+		}
+		return nil
+	})
+
+	// Applies when the Authorization header is set with the Basic scheme
+	api.BasicAuthAuth = func(user string, pass string) (*models.Principal, error) {
+		if user != "admin" || pass != "admin" {
```
Comment (renuka-fernando) [inline api-operator/pkg/envoy/server/api/restserver/configure_restapi.go:65]: Shall we have a config for user name and password

File: api-operator/pkg/envoy/server/api/restserver/configure_restapi.go [hunk 2]
```diff
@@ -0,0 +1,174 @@
+// This file is safe to edit. Once it exists it will not be overwritten
```
Comment (renuka-fernando) [inline api-operator/pkg/envoy/server/api/restserver/configure_restapi.go:1]: Shall we add license header

**Review by renuka-fernando — APPROVED** [review]
merging to feature branch 2.0.0-1

File: api-operator/pkg/envoy/server/api/models/error.go [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+			if err := m.Error[i].Validate(formats); err != nil {
+				if ve, ok := err.(*errors.Validation); ok {
+					return ve.ValidateName("error" + "." + strconv.Itoa(i))
+				}
+				return err
+			}
+		}
+
+	}
+
```
Comment (Meruja) [inline api-operator/pkg/envoy/server/api/models/error.go:96]: Shall we remove these extra lines unless this file is auto-generated?

File: api-operator/pkg/envoy/server/api/models/principal.go [hunk 4]
```diff
@@ -0,0 +1,49 @@
+// Code generated by go-swagger; DO NOT EDIT.
```
Comment (Meruja) [inline api-operator/pkg/envoy/server/api/models/principal.go:1]: Shall we add license header here?

File: api-operator/pkg/envoy/server/api/restserver/configure_restapi.go [hunk 5]
```diff
@@ -0,0 +1,174 @@
...
+		return &p, nil
+	}
+
+	// Set your custom authorizer if needed. Default one is security.Authorized()
+	// Expected interface runtime.Authorizer
+	//
+	// Example:
+	// api.APIAuthorizer = security.Authorized()
+	if api.ApIsAllGetApisHandler == nil {
+		api.ApIsAllGetApisHandler = a_p_is_all.GetApisHandlerFunc(func(params a_p_is_all.GetApisParams, principal *models.Principal) middleware.Responder {
```
Comment (Meruja) [inline api-operator/pkg/envoy/server/api/restserver/configure_restapi.go:83]: Length exceeded(120 characters). Shall we break into two lines?

File: api-operator/pkg/envoy/server/api/models/error.go (hunk 3 above)
Comment (Jayanie) [reply api-operator/pkg/envoy/server/api/models/error.go:96]: This file is auto-generated.

File: api-operator/pkg/envoy/server/api/models/principal.go (hunk 4 above)
Comment (Jayanie) [reply api-operator/pkg/envoy/server/api/models/principal.go:1]: These files are auto-generated.

File: api-operator/pkg/envoy/server/api/restserver/configure_restapi.go (hunk 5 above)
Comment (Jayanie) [reply api-operator/pkg/envoy/server/api/restserver/configure_restapi.go:83]: Fixed with 28191756937ef7cf2473f7531e68b5a81ae87e68

**Pull request merged**